src/App.tsx

```diff
@@ -1,73 +1,65 @@
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { Autocomplete } from './components/Autocomplete';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const { name, born, died } = selectedPerson || {};
+  const [delay] = useState(300);
+  const [query, setQuery] = useState('');
+  const [focus, setFocus] = useState(false);
```

Comment:
App component shouldn't know about query or focus. move logic with it to Autocomplete component.